### ISSUE TITLE
Error on Window 10

### ISSUE THREAD

**Issue by TokiWartooth101** [nwoolls/MultiMiner#322]
I Just downloaded and tried to install multi miner and run it on my get win 10 laptop however i keep getting the error when installing bfg miner. I googled a little and found that its a Windows Product Activation (WPA) requires is damaged or missing issue.

Also I tried on my Mac Pro that is currently running 10.9.5 and didnt receive any errors, but when I try to open the app the mouse spins for a while then stops and nothing happens

![multi miner error](https://user-images.githubusercontent.com/34991141/34464606-5e8c1cd6-ee3b-11e7-9680-fa2a01c21d74.jpg)

Thanks in advance

**Comment by nwoolls** [comment]
@TokiWartooth101 on Windows you will likely need to disable Windows Defender and any antivirus, or install BFGMiner yourself in `<multiminer>\Miners\BFGMiner`. On macOS, wait longer for it to start, there are troubleshooting tips on the wiki:

* https://github.com/nwoolls/MultiMiner/wiki/Installation#mac-os-x

**Comment by TokiWartooth101** [comment]
I turned both my anti virus and windows defender and tried to have MultiMiner automatically install BFGMiner, as well as tried to manually install it through the zip. The app files dont load anything up, nor does it work if i manually extract into the MultiMiner\miner folder (only pulls out txt files) 

I looked a little more on the Apple and mono is timing out

**Comment by nwoolls** [comment]
Something is still blocking the extraction of BFGMiner on your system. You will not be able to mine until you resolve that.

Mono does not "time out" that I am aware of - what is the exact output from the command listed in the troubleshooting section linked in my previous response?

**Comment by TokiWartooth101** [comment]
When using the terminal command I keep getting file not found. As for the console it has a log file for mono when I have tried to run MultiMiner. It is too long to post, I can email a copy of the file if you would like.

**Comment by nwoolls** [comment]
> When using the terminal command I keep getting file not found.

Can you copy & paste the command you are executing and the output? You may need to adjust the path to be wherever the `.app` file lives.

**Comment by TokiWartooth101** [comment]
I opened the App info and showed contents to copy & paste all with no luck. 

`michaelahanspro:~ callahan$ cd ~/Desktop/MultiMiner.app/Contents/Resources
-bash: cd: /Users/callahan/Desktop/MultiMiner.app/Contents/Resources: No such file or directory
michaelahanspro:~ callahan$ cd ~/Desktop/MultiMiner 2.app/Contents/Resources
-bash: cd: /Users/callahan/Desktop/MultiMiner: No such file or directory
michaelahanspro:~ callahan$ cd ~/Users/callahan/Desktop/MultiMiner 2.app/Contents/Resources
-bash: cd: /Users/callahan/Users/callahan/Desktop/MultiMiner: No such file or directory
michaelahanspro:~ callahan$ cd ~/Users/callahan/Desktop/MultiMiner 2.app/Contents/Resources mono --arch=32 MultiMiner.Win.exe
-bash: cd: /Users/callahan/Users/callahan/Desktop/MultiMiner: No such file or directory
michaelahanspro:~ callahan$ cd ~/Users/callahan/Desktop/MultiMiner.app/Contents/Resources mono --arch=32 MultiMiner.Win.exe
-bash: cd: /Users/callahan/Users/callahan/Desktop/MultiMiner.app/Contents/Resources: No such file or directory
michaelahanspro:~ callahan$ mono --arch=32 MultiMiner.Win.exe
Cannot open assembly 'MultiMiner.Win.exe': No such file or directory.`

I did use mono and it gave me its options list; here is a part of it

`michaelahanspro:~ callahan$ mono
Usage is: mono [options] program [program-options]`

`Development:
    --aot[=<options>]      Compiles the assembly to native code
    --debug[=<options>]    Enable debugging support, use --help-debug for details`



I can open BFG Miner on its own logging a miner in but not hashing. Looked on the forums some and found its designed to just pick up serial devices instead of just using my gpu. I looked at the read me and saw some commands relating to gpu. Are the commands needed to be loaded through the config?

**Comment by nwoolls** [comment]
It looks like you need to figure out where you have extracted `MultiMiner.app` - according to what you've pasted above it's not on your Desktop. You need to change the commands to match where you have extracted the `.app` file.

For instance, if it's your Desktop:

```
cd ~/Desktop/MultiMiner.app/Contents/Resources
```

If it's your Downloads folder:

```
cd ~/Downloads/MultiMiner.app/Contents/Resources
```

If it's your Applications folder:

```
cd /Applications/MultiMiner.app/Contents/Resources
```

etc.

**Comment by TokiWartooth101** [comment]
thats weird I moved a copy of everything to my desktop, or at least I thought

``Last login: Tue Jan  2 15:36:11 on ttys002
michaelahanspro:~ callahan$ cd ~/Downloads/MultiMiner.app/Contents/Resources
michaelahanspro:Resources callahan$ mono --arch=32 MultiMiner.Win.exe
2018-01-02 18:20:29.029 mono32[1189:d07] WARNING: CFSTR("\37777777742\37777777634\37777777626") has non-7 bit chars, interpreting using MacOS Roman encoding for now, but this will change. Please eliminate usages of non-7 bit chars (including escaped characters above \177 octal) in CFSTR().
System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x0001e] in <4fdc5ed61a074cafb49fa42deb20d521>:0 
  at MultiMiner.Utility.Net.LocalNetwork.GetWorkGroupName () [0x0001b] in <4f24c00fe8bd4bc8b8f1cb0adfe506ce>:0 
  at MultiMiner.UX.ViewModels.ApplicationViewModel.EnableRemoting () [0x0001f] in <77463dbf03f74807875d550c9f9e226b>:0 
  at MultiMiner.UX.ViewModels.ApplicationViewModel.SetupRemoting () [0x00022] in <77463dbf03f74807875d550c9f9e226b>:0 
  at MultiMiner.Win.Forms.MinerForm.SetupApplication () [0x000e7] in <1382821a0ef9444185b7d7b778bcd175>:0 
  at MultiMiner.Win.Forms.MinerForm.MainForm_Load (System.Object sender, System.EventArgs e) [0x00000] in <1382821a0ef9444185b7d7b778bcd175>:0 
  at System.Windows.Forms.Form.OnLoad (System.EventArgs e) [0x0001f] in <36fbc38531724d82b10aef6ebafea6e9>:0 
  at System.Windows.Forms.Form.OnLoadInternal (System.EventArgs e) [0x00023] in <36fbc38531724d82b10aef6ebafea6e9>:0``

```

**Comment by nwoolls** [comment]
@TokiWartooth101 looks like you enabled remoting but do not have a workgroup defined in macOS. I'll get this fixed in an update. You can work around it by defining a workgroup in macOS System Preferences > Network > Advanced > WINS.

**Comment by nwoolls** [comment]
Fixed for the next update

**Comment by TokiWartooth101** [comment]
I tried to get this apple to connect to my windows workgroup. The IP was odd ipv4: 169.254.252.238/16. It did start to show under devices but won't connect. I had to open everything up and just allow it to access my laptop as a guest. Also, I changed my laptops name and the old name is stuck in the devices area. I guess this 2006 has just about had it lol

Once I do get the workgroup communicating is there a great deal more to get an old gfx card connected to BFG Miner or MultiMiner? If there isn't a little bit of scripting it would like to skip it in favor of something simple I can learn on till I get further in school. Much appreciative of the help so far!
